api/systemNodes/initialize.js

```diff
@@ -3,14 +3,66 @@
 var self = initialize;
 module.exports = self;
 
+var async = require('async');
+
+var APIAdapter = require('../../common/APIAdapter.js');
+
 function initialize(req, res) {
   var bag = {
     reqBody: req.body,
-    resBody: {}
+    resBody: {},
+    apiAdapter: new APIAdapter(req.headers.authorization.split(' ')[1])
   };
 
   bag.who = util.format('systemNodes|%s', self.name);
   logger.info(bag.who, 'Starting');
 
-  sendJSONResponse(res, bag.resBody);
+  async.series([
+      _checkInputParams.bind(null, bag),
+      _getSystemSettings.bind(null, bag)
+    ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (err)
+        return respondWithError(res, err);
+
+      sendJSONResponse(res, bag.resBody);
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (global.config.admiralIP !== 'localhost' &&
+    global.config.admiralIP !== '127.0.0.1')
+    return next(
+      new ActErr(who, ActErr.OperationFailed, 'Unable to bring up system node')
+    );
+
+  return next();
+}
+
+function _getSystemSettings(bag, next) {
+  var who = bag.who + '|' + _getSystemSettings.name;
+  logger.verbose(who, 'Inside');
+
+  bag.apiAdapter.getSystemSettings('',
+    function (err, systemSettings) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Failed to get system settings : ' + util.inspect(err))
+        );
+
+      if (systemSettings.runMode !== 'dev')
```

Comment:
ok. actually, we dont even need to do the runmode check. please see my next comment. this route behavior(and its functionality) should not change because of runmode.